### ISSUE TITLE
Use state values instead of settings for ephemeral values

### ISSUE THREAD

**Issue by bradbulger** [backdrop-contrib/notify#10]
I've noticed that the Notify settings file is constantly changing, because it's storing things like the timestamp of the next cron job, the number of notifications sent in the last batch, etc. Those aren't configuration values, really. state_set() and state_get() are more appropriate ways to store and reference that information, I think.

**Comment by Graham-72** [comment]
Yes. I will make some changes once I have assessed the purpose of all the data variables.

**Comment by bradbulger** [comment]
I have that done, I think. Sorry, I meant to follow up the creation of this with a pull request, but got called away by other issues. The ones that I think should change - because they're modified by each cron run - are:

notify_cron_next
notify_num_sent
notify_num_failed
notify_send_last
notify_send_start
notify_users

**Comment by Graham-72** [comment]
@bradbulger I think this is fixed now. Please can you check it before I make a new release?

**Comment by bradbulger** [comment]
I tried out those changes, they seem to work. I couldn't figure out how to run notify.test. The tests for comment_notify and node_notify mostly ran fine, and the few failures were unrelated to these changes. I added an update hook to move values from config settings to state and then remove those keys from the config.
